frontend/src/metabase/entities/containers/rtk-query/EntityListLoaderRtkQuery.tsx

```diff
@@ -0,0 +1,332 @@
+
+  useEffect(() => {
+    if (rtkError) {
+      // @ts-expect-error - invalid typings in redux-actions package
+      dispatch(setRequestError(requestStatePath, queryKey, rtkError));
+    }
+  }, [dispatch, rtkError, requestStatePath, queryKey]);
+
+  useEffect(() => {
+    if (data) {
+      const { results, metadata } = transformResponse(data);
+
+      if (!Array.isArray(results)) {
+        throw new Error(`Invalid response listing ${entityDefinition.name}`);
+      }
+
+      const normalized = entityDefinition.normalizeList(results);
+
+      const result = {
+        ...normalized,
+        metadata,
+        entityQuery,
+      };
+
+      dispatch({
+        type: entityDefinition.actionTypes.FETCH_LIST,
+        payload: result,
+      });
+
+      // NOTE Atte Keinänen 8/23/17:
+      // Dispatch `setRequestLoaded` after clearing the call stack because we want to the actual data to be updated
+      // before we notify components via `state.requests.fetches` that fetching the data is completed
+      // @ts-expect-error - invalid typings in redux-actions package
+      setTimeout(() => dispatch(setRequestLoaded(requestStatePath, queryKey)));
+    }
+  }, [
+    dispatch,
+    data,
+    entityDefinition,
+    entityQuery,
+    requestStatePath,
+    queryKey,
+  ]);
+
+  const onLoadedRef = useLatest(onLoaded);
+
+  useEffect(() => {
+    if (data) {
+      onLoadedRef.current?.(data);
+    }
+  }, [data, onLoadedRef]);
+
+  // merge props passed in from stacked Entity*Loaders:
+  const allError = error || (allErrorProp == null ? null : allErrorProp);
+  const allFetched =
+    fetched && (allFetchedProp == null ? true : allFetchedProp);
+  const allLoaded = loaded && (allLoadedProp == null ? true : allLoadedProp);
+  const allLoading =
+    loading || (allLoadingProp == null ? false : allLoadingProp);
+  const finalListName = listName || entityDefinition.nameMany;
+
+  const children = (
+    <ComposedComponent
+      {...props}
+      {...{
+        [finalListName]: wrappedList,
+        [`reload${capitalize(finalListName)}`]: refetch,
+      }}
+      allError={allError}
+      allFetched={allFetched}
+      allLoaded={allLoaded}
+      allLoading={allLoading}
+      entityQuery={entityQuery}
+      error={error}
+      fetched={fetched}
+      list={wrappedList}
+      loaded={loaded}
+      loading={loading || isFetching}
+      metadata={metadata}
+      reload={refetch}
+    />
+  );
+
+  if (loadingAndErrorWrapper) {
+    return (
+      <LoadingAndErrorWrapper
+        loading={!allFetched || isLoading}
+        error={allError}
+        noWrapper
+      >
+        {children}
+      </LoadingAndErrorWrapper>
+    );
+  }
+
+  return children;
+}
+
+/**
+ * @deprecated HOCs are deprecated
+ */
+export const entityListLoaderRtkQuery =
+  (eolProps: any) =>
+  (ComposedComponent: (props: any) => ReactNode) =>
+  // eslint-disable-next-line react/display-name
+  (props: any): ReactNode => (
```

Comment:
Should we give this component a name and get rid of the `eslint-disable-next-line` directive?

Reply:
828292b2b3aeb159fcf43ee4b5b5fa8ac3dfb325